reyna/src/com/b2msolutions/reyna/services/BlackoutTime.java

```diff
@@ -0,0 +1,60 @@
+package com.b2msolutions.reyna.services;
+
+import android.content.Context;
+import com.b2msolutions.reyna.Preferences;
+import com.b2msolutions.reyna.Time;
+import com.b2msolutions.reyna.TimeRange;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.*;
+
+public class BlackoutTime {
+
+    private final Context context;
+
+    public BlackoutTime(Context context) {
+        this.context = context;
+    }
+
+    public boolean canSendOnWlan(Calendar now) throws ParseException {
```

Comment:
I suggest to move canSendOnWlan and canSendOnWwan out of this class and make this class has single responsibility to check time.

@Youhana-Hana